### PR TITLE
Resolve known version for locale conflicts

### PR DESCRIPTION
**Pull request by kevindew**
This is a migration which removes items which have version conflicts. Inside the migration are details about all the items which would be deleted from the database.

Outputs from running locally
```
vm  publishing-api git:(nuke-invalid-data) ✗ rake duplicate_content_items:version_for_locale

There are 133 version for locale conflicts 😭
vm  publishing-api git:(nuke-invalid-data) ✗ rake db:migrate

== 20160825141159 ResolveVersionForLocaleConflicts: migrating =================
== 20160825141159 ResolveVersionForLocaleConflicts: migrated (1.4831s) ========

vm  publishing-api git:(nuke-invalid-data) ✗ rake duplicate_content_items:version_for_locale

There are 0 version for locale conflicts 😀
```

It can be recovered, I have a sql file with all of the items to be deleted. Although it could be dangerous to run that in full as it may introduce items that have been deleted elsewhere.

```
vm  publishing-api git:(nuke-invalid-data) ✗ psql -d publishing_api_development -f version-for-locale-recovery.sql

vm  publishing-api git:(nuke-invalid-data) ✗ rake db:rollback

== 20160825141159 ResolveVersionForLocaleConflicts: reverting =================
== 20160825141159 ResolveVersionForLocaleConflicts: reverted (0.0000s) ========

vm  publishing-api git:(nuke-invalid-data) ✗ rake duplicate_content_items:version_for_locale

There are 133 version for locale conflicts 😭
```

### CODE REVIEW

File: db/migrate/helpers/delete-content-item.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+module Helpers
+  module DeleteContentItem
+    def self.destroy_supporting_objects(content_items)
```
Comment (dougdroper) [inline db/migrate/helpers/delete-content-item.rb]: Nice, can we update the readme to point to this? 
https://github.com/alphagov/publishing-api#deleting-content-items

Comment (dougdroper) [inline db/migrate/helpers/delete-content-item.rb]: And can we `Array` the content_items?

Comment (kevindew) [inline db/migrate/helpers/delete-content-item.rb]: Good point about the readme

hmm the `Array` would be unnecessary as ActiveRecord handles it being a model or an array. Do you mean to add that so it's explicit that it accepts a single item or a collection?

Comment (dougdroper) [inline db/migrate/helpers/delete-content-item.rb]: explicit that it accepts a single item or a collection

File: db/migrate/20160825141159_resolve_version_for_locale_conflicts.rb [hunk 2]
```diff
@@ -0,0 +1,690 @@
...
+      delete(content_item) if content_item
+    end
+  end
+
+  def delete(content_item)
+    Helpers::DeleteContentItem.destroy_supporting_objects(content_item)
+    content_item.destroy
+  end
+
+  def to_delete
```
Comment (cbaines) [inline db/migrate/20160825141159_resolve_version_for_locale_conflicts.rb]: It would be good to comment here saying where the data below came from.

**Comment by danielroseman** [comment]
:+1:, merge once conflicts are resolved

**Comment by kevindew** [comment]
Thanks @danielroseman - Since I want these commits to sit before the base_pase_for_state_conflicts (so validations are in chronological order and latest validation is in db/schema.rb) I've got with a merge commit. Do shout if you think I should rebase it instead.

**Pull request merged**